core/src/main/java/org/mule/runtime/core/privileged/processor/chain/DefaultMessageProcessorChainBuilder.java

```diff
@@ -304,14 +305,19 @@ public interface MessagingExceptionHandlerAware {
     void setMessagingExceptionHandler(FlowExceptionHandler messagingExceptionHandler);
   }
 
+  public interface SpanCustomizationInfoAware {
```

Comment:
I would set this as a not inner interface. Add a jdoc when it is in its own file.